core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go

```diff
@@ -448,6 +449,14 @@ func getStarlarkFilesArtifacts(composeVolumes []types.ServiceVolumeConfig, servi
 		var shouldPersist bool
 		switch volumeType {
 		case types.VolumeTypeBind:
+			// Handle case where home path is reference
```

Comment:
The `~` home folder is a strong indication the volume bind might be outside the project folder. But a lot of other paths can also be outside. Wondering is there a better way of detecting it e.g. resolving volume paths against the project path and pattern matching it?  Maybe we could just block absolute paths (`~` includude) or relative paths going upstream `../something`

Reply:
Good point, right now, we are treating absolute paths as persistent directories, but that doesn't account for home path and didn't think about the case where relative paths that go upstream.

Reply:
For now, I'm going to keep the absolute path behavior (these become persistent behavior), but I'm going to account for the case where relative paths goes upstream eg. `..` and block that here as well.

Reply:
I agree this can be handled better, but want to leave that for the future if this feature requires iteration from usage : )